api/lib/infrastructure/utils/url-utils.js

```diff
@@ -0,0 +1,87 @@
+import showdown from 'showdown';
+import urlRegex from 'url-regex-safe';
+import _ from 'lodash';
+import { logger } from '../logger.js';
+import { CookieJar } from 'tough-cookie';
+import { wrapper } from 'axios-cookiejar-support';
+import axios from 'axios';
+
+export class UrlUtils {
+  static findUrlsInMarkdown(value) {
+    const safeValue = value || '';
+    const converter = new showdown.Converter();
+    const html = converter.makeHtml(safeValue);
+    const urls = html.match(urlRegex({ strict: true }));
+    if (!urls) {
+      return [];
+    }
+    return _.uniq(urls.map(cleanUrl).map(prependProtocol));
+  }
+
+  // identifiedUrls --> [{ id: <someId>, url: <theUrl>}, ...]
```

Comment:
N'ayez pas peur de la JSDoc.

```suggestion
  /**
   * @param {{ id: string, url: string }[]} identifiedUrls
   * @returns {{
   *   id: string,
   *   url: string,
   *   status: string,
   *   error: string,
   *   comments: string,
   * }[]}
   */
```